src/test/java/bio/terra/service/iam/AccessTest.java

```diff
@@ -253,6 +264,56 @@ public void fileAclTest() throws Exception {
         assertFalse("Discoverer can not read the file", canReadBlob(discovererStorage, blobId));
     }
 
+    @Test
+    public void checkCustodianPermissions() throws  Exception {
+        IngestRequestModel request = dataRepoFixtures.buildSimpleIngest(
+            "participant", "ingest-test/ingest-test-participant.json");
+        dataRepoFixtures.ingestJsonData(steward(), datasetId, request);
+
+        request = dataRepoFixtures.buildSimpleIngest(
+            "sample", "ingest-test/ingest-test-sample.json");
+        dataRepoFixtures.ingestJsonData(steward(), datasetId, request);
+
+        DatasetModel dataset = dataRepoFixtures.getDataset(steward(), datasetId);
+        String datasetBqSnapshotName = "datarepo_" + dataset.getName();
+
+        BigQuery custodianBigQuery = BigQueryFixtures.getBigQuery(dataset.getDataProject(), custodianToken);
+        try {
+            BigQueryFixtures.datasetExists(custodianBigQuery, dataset.getDataProject(), datasetBqSnapshotName);
+            fail("custodian shouldn't be able to access bq dataset before it is shared with them");
+        } catch (IllegalStateException e) {
+            assertThat("checking message for pdao exception error",
+                e.getMessage(),
+                equalTo("existence check failed for " + datasetBqSnapshotName));
+        }
+
+        dataRepoFixtures.addDatasetPolicyMember(
+            steward(),
+            datasetId,
+            IamRole.CUSTODIAN,
+            custodian().getEmail());
+        DataRepoResponse<EnumerateDatasetModel> enumDatasets = dataRepoFixtures.enumerateDatasetsRaw(custodian());
+        assertThat("Custodian is authorized to enumerate datasets",
+            enumDatasets.getStatusCode(),
+            equalTo(HttpStatus.OK));
+
+        boolean custodianHasAccess =
+            BigQueryFixtures.hasAccess(custodianBigQuery, dataset.getDataProject(), datasetBqSnapshotName);
+
+        // gets the "sample" table and makes a table ref to use in the query
+        String tableRef = BigQueryFixtures.makeTableRef(dataset, dataset.getSchema().getTables().get(1).getName());
+        String sql = String.format("SELECT * FROM %s LIMIT %s", tableRef, 1000);
+        TableResult results = BigQueryFixtures.query(sql, custodianBigQuery);
+        Assert.assertEquals(7, results.getTotalRows());
+
+        assertThat("custodian can access the bq snapshot after it has been shared",
+            custodianHasAccess,
+            equalTo(true));
+
+        SnapshotSummaryModel snapshotSummaryModel =
+            dataRepoFixtures.createSnapshot(custodian(), datasetSummaryModel, "ingest-test-snapshot.json");
```

Comment:
Are you testing anything specifically with this snapshot creation? Fine to leave it in, just curious whether there was something in particular you were testing.